content/general/concepts/abstract-factory/abstract-factory.md

```diff
@@ -0,0 +1,396 @@
+---
+Title: 'Abstract Factory Pattern'
+Description: 'The abstract factory pattern relies on a common interface for multiple factories and defers instantiation logic to their concrete model classes.'
+Subjects:
+  - 'Computer Science'
+  - 'Interview Prep'
+Tags:
+  - 'Classes'
+  - 'Conceptual'
+  - 'Constructors'
+  - 'Objects'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The abstract factory is often thought of as a factory of factory patterns. Similar to the factory pattern, the abstract factory relies on a common interface and defers instantiation logic to its concrete model classes. However, it covers multiple factories. Additionally, unlike the factory pattern, the abstract factory is typically built using composition whereas the factory is not.
```

Comment:
Yes. Being super honest, I didn't really understand this too much, and was hoping it would be missed during PR.

The idea is about composition vs inheritance. Inheritance, you get stuff from a parent class (either bits of functionality or useful fields) and composition (I think) you get what the object needs by having properties containing other useful objects. I.e. when then objects created / composed, it has stuff provided by the constructor??

I'm super unsure of this!! Will have to read more :)

Reply:
```suggestion
The abstract factory pattern utilizes a common interface for multiple factories without defining their concrete model classes. It is often thought of as a factory of factory patterns where object composition is used to separate instantiation logic from application logic.
```

What about this?